install.sh

```diff
@@ -1,90 +1,349 @@
+show_loader() {
+    local message=$1
+    local pid=$!
+    local delay=0.3
+    local spinstr='|/-\\'
+    while [ "$(ps a | awk '{print $1}' | grep $pid)" ]; do
+        local temp=${spinstr#?}
+        printf "\r%s [%c]  " "$message" "$spinstr"
+        local spinstr=$temp${spinstr%"$temp"}
+        sleep $delay
+    done
+    printf "\r%s [Done]   \n" "$message"
 }
 
-echo "##########################################\nInstalling dependencies\n##########################################"
-# Install dependencies
-if ! command -v node &> /dev/null
-then
-    echo "node could not be found. Please install Node.js 18 or above."
+# Check for Node.js
+echo "Checking for Node.js..."
+if ! command -v node &> /dev/null; then
+    echo "Node.js could not be found. Please install node js 18 or above."
     exit 1
 fi
+
+# Verify Node.js version
 version=$(node -v | cut -d'.' -f1 | tr -d 'v')
-if [ "$version" -lt 18 ]
-then
-  echo "Invalid Node.js version. Expected 18 or above, but got $version."
-  exit 1
+if [ "$version" -lt 18 ]; then
+    echo "Invalid Node.js version. Expected 18 or above, but got $version."
+    exit 1
 fi
+echo "Node.js version is valid."
+
+# Check and Install AWS CLI
+echo "Checking for AWS CLI..."
 if aws --version; then
-  echo "##########################################\nAWS CLI Installed\n##########################################"
+    echo "AWS CLI is already installed."
 else
-  curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
-  sudo installer -pkg AWSCLIV2.pkg -target /
+    echo "Installing AWS CLI..."
+    curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+    sudo installer -pkg AWSCLIV2.pkg -target / & show_loader "Installing AWS CLI.."
+    echo "AWS CLI is installed."
 fi
-if ! command -v aws &> /dev/null
-then
-    echo "AWS CLI could not be found. Please rerun \`sh install.sh\` with Sudo access"
+
+# Check if AWS CLI installation was successful
+if ! command -v aws &> /dev/null; then
+    echo "AWS CLI could not be found. Please rerun 'sh install.sh' with Sudo access."
     exit 1
 fi
-npm install -g aws-cdk
-if ! command -v cdk &> /dev/null
-then
-    echo "AWS CDK could not be found. Please rerun \`sh install.sh\` with Sudo access"
+
+# Install AWS CDK
+echo "Installing AWS CDK..."
+npm install -g aws-cdk & show_loader "Installing AWS CDK..."
+echo "AWS CDK is installed successfully."
+
+# Check for AWS CDK
+if ! command -v cdk &> /dev/null; then
+    echo "AWS CDK could not be found. Please rerun 'sh install.sh' with Sudo access."
     exit 1
 fi
-cdk --version
+
+# Determine OS and run respective dependency script
 os=$(uname)
-if [ "$os" = "Linux" ]; then
-  sh linux_deps.sh
-elif [ "$os" = "Darwin" ]; then
-  sh mac_deps.sh
-else
-  echo "Unsupported operating system."
-  exit 1
-fi
+case "$os" in
+  "Linux")
+    echo "Detecting operating system: Linux"
+    (sh linux_deps.sh & show_loader "Running Linux dependencies script...")
```

Comment:
Also, any specific reason for running/suggesting `sh <filename>.sh` over `bash <filename>.sh`?
I'd suggest using `bash` everywhere, since it's available on most systems nowadays, while `sh` could be a link to any shell (`ash`, `dash`, etc.) and the difference in behavior from `bash` is not something we'd want to handle.

Reply:
@jagan-jaya Could you please let us know if there is any specific reason we are using `sh <filename>.sh`